Xamarin.Forms.Platform.Android/Renderers/ShellFlyoutRecyclerAdapter.cs

```diff
@@ -59,7 +59,10 @@ public override int GetItemViewType(int position)
 			var template = dataTemplate.SelectDataTemplate(item.Element, Shell);
 			var id = ((IDataTemplateController)template).Id;
 
-			_templateMap[id] = template;
+			if (_templateMap.ContainsKey(id))
+				_templateMap[id] = template;
+                        Else
+                                _templateMap.Add(id) = template;
```

Comment:
There are some code format issues, could you check it?. 
On the other hand, although not essential, could you include a sample in Core Gallery to validate the issue?